### PR TITLE
Improve API docs

### PR DESCRIPTION
**Pull request by kaneeldias**
## Purpose
$subject

Fixes: https://github.com/ballerina-platform/ballerina-standard-library/issues/2585

## Checklist
- [x] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

**Review by daneshk — APPROVED** [review]

**Review by BuddhiWathsala — APPROVED** [review]

File: ballerina/Module.md [hunk 1]
```diff
@@ -70,10 +70,10 @@ jdbc:Client|sql:Error dbClient = new (
 );
 ```
 
-The `dbClient` receives some custom properties within the
-[jdbc:Options](https://docs.central.ballerina.io/ballerinax/java.jdbc/latest/records/Options),   
+The `jdbc:Client` receives some custom properties within the
+[`jdbc:Options`](https://docs.central.ballerina.io/ballerinax/java.jdbc/latest/records/Options),
```
Comment (praneesha) [inline ballerina/Module.md:74]:
```suggestion
[`jdbc:Options`](https://docs.central.ballerina.io/ballerinax/java.jdbc/latest/records/Options)
```

File: ballerina/Module.md [hunk 2]
```diff
@@ -199,15 +197,15 @@ sql:ParameterizedQuery query1 = ` WHERE id < ${id} AND age > ${age}`;
 sql:ParameterizedQuery sqlQuery = sql:queryConcat(query, query1);
 ```
 
-The query with the `IN` operator can be created using the `sql:ParameterizedQuery` like below. Here you need to flatten the array and pass each element separated by a comma.
+The query with the `IN` operator can be created using the `sql:ParameterizedQuery` as shown below. Here you need to flatten the array and pass each element separated by a comma.
```
Comment (praneesha) [inline ballerina/Module.md:200]:
```suggestion
The query with the `IN` operator can be created using the `sql:ParameterizedQuery` as shown below. Here, you need to flatten the array and pass each element separated by a comma.
```

File: ballerina/Module.md [hunk 3]
```diff
@@ -199,15 +197,15 @@ sql:ParameterizedQuery query1 = ` WHERE id < ${id} AND age > ${age}`;
...
 
 ```ballerina
 int[] ids = [1, 2, 3];
 sql:ParameterizedQuery query = `SELECT count(*) as total FROM DataTable 
-                                WHERE row_id in (${ids[0]}, ${ids[1]}, ${ids[2]})`
+                                WHERE row_id IN (${ids[0]}, ${ids[1]}, ${ids[2]})`
 ```
 
-The util function `sql:arrayFlattenQuery()` is introduced to make the array flatten easier. It makes the inclusion of varying array elements into the query easier by flattening the array to return a parameterized query. You can construct the complex dynamic query with the `IN` operator by using both functions like below.
+The util function `sql:arrayFlattenQuery()` is used to make the array flatten easier. It makes the inclusion of varying array elements into the query easier by flattening the array to return a parameterized query. You can construct the complex dynamic query with the `IN` operator by using both functions as shown below.
```
Comment (praneesha) [inline ballerina/Module.md:208]:
```suggestion
The `sql:arrayFlattenQuery()` util function is used to make the array flatten easier. It makes the inclusion of varying array elements into the query easier by flattening the array to return a parameterized query. You can construct the complex dynamic query with the `IN` operator by using both functions as shown below.
```

File: ballerina/Module.md [hunk 4]
```diff
@@ -218,19 +216,20 @@ sql:ParameterizedQuery sqlQuery =
 
 #### Creating Tables
 
-This sample creates a table with two columns. One column is of type `int` and the other is of type `varchar`.
+This sample creates a table with three columns. The first column is a primary key of type `int`,
```
Comment (praneesha) [inline ballerina/Module.md:219]:
```suggestion
This sample creates a table with three columns. The first column is a primary key of type `int`
```

File: ballerina/Module.md [hunk 5]
```diff
@@ -218,19 +216,20 @@ sql:ParameterizedQuery sqlQuery =
...
+while the second column is of type `int`, and the other is of type `varchar`.
```
Comment (praneesha) [inline ballerina/Module.md:220]:
```suggestion
while the second column is of type `int` and the other is of type `varchar`.
```

File: ballerina/Module.md [hunk 6]
```diff
@@ -301,11 +302,11 @@ First, a type is created to represent the returned result set. This record can b
 according to the requirement. If an open record is defined, the returned stream type will include both defined fields
 in the record and additional database columns fetched by the SQL query which are not defined in the record.
 Note the mapping of the database column to the returned record's property is case-insensitive if it is defined in the
-record(i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional Column names
-added to the returned record as in the SQL query. If the record is defined as a close record, only defined fields in the
+record(i.e., the `ID` column in the result can be mapped to the `id` property in the record). Additional column names
+added to the returned record as in the SQL query. If the record is defined as a closed record, only defined fields in the
```
Comment (praneesha) [inline ballerina/Module.md:306]:
```suggestion
added to the returned record as in the SQL query. If the record is defined as a closed record, only defined fields in the
```
```suggestion
are added to the returned record as in the SQL query. If the record is defined as a closed record, only defined fields in the
```

File: ballerina/client.bal [hunk 7]
```diff
@@ -55,11 +55,11 @@ public isolated client class Client {
     } external;
 
     # Executes the query, which is expected to return at most one row of the result.
-    # If the query does not return any results, `sql:NoRowsError` is returned
+    # If the query does not return any results, `sql:NoRowsError` is returned.
```
Comment (praneesha) [inline ballerina/client.bal:58]:
```suggestion
    # If the query does not return any results, an`sql:NoRowsError` is returned.
```

File: ballerina/client.bal [hunk 8]
```diff
@@ -110,32 +110,33 @@ public isolated client class Client {
...
+# Constants to represent database operations.
 public enum Operations {
     NONE,
     EXECUTE,
     BATCH_EXECUTE,
     ALL
 }
 
-# Provides a set of configurations for the JDBC Client to be passed internally within the module.
+# Provides an additional set of configurations for the JDBC Client to be passed internally within the module.
```
Comment (praneesha) [inline ballerina/client.bal:132]:
```suggestion
# An additional set of configurations for the JDBC Client to be passed internally within the module.
```

File: docs/spec/spec.md [hunk 9]
```diff
@@ -3,14 +3,14 @@
...
-_Updated_: 2022/01/14  
+_Updated_: 2022/01/25  
 _Issue_: [#2290](https://github.com/ballerina-platform/ballerina-standard-library/issues/2290)
 
 # Introduction
 
-This is the specification for JDBC standard library, which provides the functionality that is required to access and 
-manipulate data stored in any type of relational database, which is accessible via Java Database Connectivity (JDBC) 
-in the [Ballerina programming language](https://ballerina.io/), which is an open-source programming
+This is the specification for the JDBC standard library, which provides the functionality that is required to access and
```
Comment (praneesha) [inline docs/spec/spec.md:11]:
```suggestion
This is the specification for the JDBC standard library, which provides the functionality that is required to access and 
```
```suggestion
This is the specification for the JDBC standard library, which provides the functionality that is required to access and 
```
This sentence is too long. Consider breaking it into multiple  sentences.

File: docs/spec/spec.md [hunk 10]
```diff
@@ -32,10 +32,10 @@ MSSQL, Postgresql and OracleDB.
 2. Executes the query, which is expected to return at most one row of the result.
 3. Executes the SQL query. Only the metadata of the execution is returned.
 4. Executes the SQL query with multiple sets of parameters in a batch. Only the metadata of the execution is returned.
-5. Executes a SQL query, which calls a stored procedure. This can either return results or nil.
+5. Executes an SQL query, which calls a stored procedure. This can either return results or nil.
 
 All the above operations make use of `sql:ParameterizedQuery` object, backtick surrounded string template to pass
```
Comment (praneesha) [inline docs/spec/spec.md:37]:
```suggestion
All the above operations make use of the sql:ParameterizedQuery` object and the string template surrounded by backticks to pass
```

File: docs/spec/spec.md [hunk 11]
```diff
@@ -62,12 +61,11 @@ public isolated function init(string url, string? user = (), string? password =
...
 * Connection properties:
    ```ballerina
-   # Provides a set of configuration related to database.
+   # Provides an additional set of configurations related to a database connection.
    #
    # + datasourceName - The driver class name to be used to get the connection
-   # + properties - The database properties which should be applied 
-   # + requestGeneratedKeys - The database operations for which generated keys should be 
-   #                          returned (NONE/EXECUTE/BATCH_EXECUTE/ALL)
+   # + properties - The database properties which should be applied when getting the connection
```
Comment (praneesha) [inline docs/spec/spec.md:67]:
```suggestion
   # + properties - The database properties which should be applied when getting the connection
```
```suggestion
   # + properties - The database properties, which should be applied when getting the connection
```

File: docs/spec/spec.md [hunk 12]
```diff
@@ -86,15 +84,15 @@ Once all the database operations are performed, the client can be closed by invo
...
+   #
+   # + return - Possible error when closing the client
+   public isolated function close() returns Error?;
    ```
 
 # 3. Queries and Values
 
-All the generic `sql` Queries and Values are supported. For more information, see the 
+All the generic `sql` queries and values are supported. For more information, see the 
 [SQL Specification](https://github.com/ballerina-platform/module-ballerina-sql/blob/master/docs/spec/spec.md#3-queries-and-values)
```
Comment (praneesha) [inline docs/spec/spec.md:96]:
```suggestion
[SQL Specification](https://github.com/ballerina-platform/module-ballerina-sql/blob/master/docs/spec/spec.md#3-queries-and-values).
```

File: ballerina/Module.md (hunk 1 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in fc94727

File: ballerina/Module.md (hunk 4 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in fc94727

File: ballerina/Module.md (hunk 5 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in fc94727

File: docs/spec/spec.md (hunk 9 above)
Comment (kaneeldias) [reply docs/spec/spec.md:11]: Fixed in fc94727

File: docs/spec/spec.md (hunk 10 above)
Comment (kaneeldias) [reply docs/spec/spec.md]: Fixed in fc94727

File: docs/spec/spec.md (hunk 11 above)
Comment (kaneeldias) [reply docs/spec/spec.md]: Fixed in fc94727

File: docs/spec/spec.md [hunk 13]
```diff
@@ -77,7 +75,7 @@ public isolated function init(string url, string? user = (), string? password =
 
 ## 2.1. Connection Pool Handling
 
-Connection Pool Handling is generic and implemented through `sql` module. For more information, see the 
+Connection pool handling is generic and implemented through `sql` module. For more information, see the
```
[the remark this answers is not in the document]
Comment (kaneeldias) [reply docs/spec/spec.md]: Fixed in fc94727

File: docs/spec/spec.md (hunk 12 above)
Comment (kaneeldias) [reply docs/spec/spec.md]: Fixed in fc94727

File: ballerina/Module.md [hunk 14]
```diff
@@ -218,19 +216,20 @@ sql:ParameterizedQuery sqlQuery =
...
+// Create the ‘Students’ table with the ‘id’, ‘name‘, and ‘age’ fields.
 sql:ExecutionResult result = 
                 check dbClient->execute(`CREATE TABLE student (
                                            id INT AUTO_INCREMENT,
                                            age INT, 
                                            name VARCHAR(255), 
                                            PRIMARY KEY (id)
                                          )`);
-//A value of the sql:ExecutionResult type is returned for 'result'. 
+// A value of the sql:ExecutionResult type is returned for 'result'.
```
[the remark this answers is not in the document]
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in fc94727

File: ballerina/Module.md (hunk 6 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in fc94727

File: ballerina/client.bal (hunk 7 above)
Comment (kaneeldias) [reply ballerina/client.bal]: Fixed in fc94727

File: ballerina/client.bal (hunk 8 above)
Comment (kaneeldias) [reply ballerina/client.bal]: Fixed in fc94727

File: ballerina/Module.md (hunk 2 above)
Comment (kaneeldias) [reply ballerina/Module.md]: Fixed in fc94727

**Pull request merged**